### PR TITLE
Drink backend: allow for arbitrary runtime

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
Make `drink_client::Client` generic over runtime and allow choosing runtime in the e2e test macro.

### CODE REVIEW

File: integration-tests/e2e-runtime-only-backend/lib.rs [hunk 1]
```diff
@@ -73,5 +73,22 @@ pub mod flipper {
 
             Ok(())
         }
+
+        /// Just instantiate a contract using non-default runtime.
+        #[ink_e2e::test(backend = "runtime-only", runtime = drink::runtime::MinimalRuntime)]
```
Comment (SkymanOne) [inline integration-tests/e2e-runtime-only-backend/lib.rs:78]: Can't you do?
```suggestion
        #[ink_e2e::test(backend = "runtime-only", runtime = ::ink_e2e::MinimalRuntime)]
```

**Review by SkymanOne — APPROVED** [review]

File: integration-tests/e2e-runtime-only-backend/lib.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply integration-tests/e2e-runtime-only-backend/lib.rs]: oh sure, that's why I reexported it (but somehow I forgot about it here...); fixed

File: crates/e2e/src/drink_client.rs [hunk 2]
```diff
@@ -221,10 +227,12 @@ impl<
             }
         };
 
+        let code_hash_raw: [u8; 32] = result.code_hash.as_ref().try_into().expect("");
```
Comment (ascjones) [inline crates/e2e/src/drink_client.rs:230]: Empty string in `expect("")`

**Review by ascjones — APPROVED** [review]

File: crates/e2e/src/drink_client.rs (hunk 2 above)
Comment (pmikolajczyk41) [reply crates/e2e/src/drink_client.rs]: must have missed it; fixed

**Pull request merged**